### PR TITLE
Add npm packages release workflow 

### PR DESCRIPTION
**Pull request by LEAFERx**
# Goals of PR

Core changes:

- add [changesets](https://github.com/changesets/changesets) package versioning and publishing management
- adjust `contracts` package for packing and publishing
  - the `contracts` package will now contain all the contracts and artifacts when published
- add `packages/sdk` package that will generate typechain bindings from the `contracts` package, and compile the ts files when published

Notes:
- all packages are currently manually published using snapshots (i.e. version should look like `0.0.0-develop-*`) with npm tag as `develop`
- duplicate of #174 but trying to merge to `develop` instead

TODOs:
- [x] automate snapshot publish using github actions

### CODE REVIEW

File: .github/workflows/release-snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Release snapshot
+
+# Releases a snapshot release when new commits merge to develop
+# This ensures the release process is working as expected as well as always gives us a release
+# we should move to a single branch in near future
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
```
Comment (MoonShiesty) [inline .github/workflows/release-snapshot.yml:10]: > Releases a snapshot release when new commits merge to **develop**

Are we releasing a snapshot on commits to develop or the main branch?

File: .github/workflows/release-snapshot.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          version: nightly
+
+      - name: Set deployment token
+        run: npm config set '//registry.npmjs.org/:_authToken' "${NPM_TOKEN}"
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Publish snapshots
+        uses: seek-oss/changesets-snapshot@v0
```
Comment (MoonShiesty) [inline .github/workflows/release-snapshot.yml:38]: do we need to build the package before we publish the snapshot?

 ```
        with:
          pre-publish: pnpm build
```

Comment (LEAFERx) [reply .github/workflows/release-snapshot.yml:38]: it is built in `prepublishOnly` npm hook

File: .github/workflows/release-snapshot.yml (hunk 1 above)
Comment (LEAFERx) [reply .github/workflows/release-snapshot.yml]: fixed

**Pull request merged**